docs/core/integration-quality-scale/rules/iqs038.md

```diff
@@ -0,0 +1,29 @@
+---
+title: "IQS038 - Make effective use of the config entry data and options"
+---
+
+IQS038 is as follows:
+
+Make effective use of the configuration entry data and options
+
+## Reasoning
+
+Configuration entries have 2 fields that are persisted, `data` and `options`.
+`data` should be used for the data that is required for the connection to be made, while `options` is used for optional data that can be used to configure the integration.
+
+Ideally, the `data` field should only be changed during a config, reauthentication or reconfigure flow.
+The `options` field can be changed during the options flow.
```

Comment:
I actually had no real good reason why we want this and what "effective" means, so this is literally just some bullshit that came into my head 😂 👌🏻